### PR TITLE
runtime-rs: add the missing default trait

### PR DESCRIPTION
**Pull request by Richardhongyu**
Some structs in the runtime-rs don't implement Default trait. This commit adds the missing Default.

Fixes: #5463.

Signed-off-by: Li Hongyu <lihongyu1999@bupt.edu.cn>

### CODE REVIEW

**Comment by studychao** [comment]
Welcome!

**Comment by Ji-Xinyou** [comment]
Apology for the ambiguity in the issue's description. IMO, the intension of adding `Default` trait was for convenience when testing.

With `Default` trait, you can construct an default one, and only test some of the fields in the struct. Therefore, my origin intension was to add `Default` trait for those **structs that need to be tested**.

Also, there are some structs like `VSock` should not have default trait since a default one is almost never valid.

**Comment by Richardhongyu** [comment]
I see your point. In this case, should I also change the test code after the `Default` trait is implemented?

**Comment by Ji-Xinyou** [comment]
> I see your point. In this case, should I also change the test code after the `Default` trait is implemented?

You could do that for sure:)

**Comment by Richardhongyu** [comment]
I just found an example and wonder whether this matches your requirements.

**Comment by Richardhongyu** [comment]
Ok, just use this example to check whether I understand your point right. I will try to fix the others. Thanks.

**Comment by Ji-Xinyou** [comment]
> Ok, just use this example to check whether I understand your point right. I will try to fix the others. Thanks.

But IMO, `default` is not only used in test codes, so I think it should be minimized. For example, the fields should be empty if possible.

**Comment by Richardhongyu** [comment]
I find other examples and keep fields empty if possible.

**Comment by Richardhongyu** [comment]
Ping@Ji-Xinyou .

File: src/libs/kata-types/src/config/agent.rs
Comment (Ji-Xinyou) [inline src/libs/kata-types/src/config/agent.rs:90]: What about making the constants here a variable, like
`const AGENT_DIAL_TIMEOUT_MS: u32 = 10;`

File: src/runtime-rs/crates/agent/src/types.rs
Comment (Ji-Xinyou) [inline src/runtime-rs/crates/agent/src/types.rs:187]: Same above

File: src/runtime-rs/crates/shim/src/args.rs
Comment (Ji-Xinyou) [inline src/runtime-rs/crates/shim/src/args.rs:120]: im not seeing the reason of this modification, but it is fine :)

**Review by Ji-Xinyou — APPROVED** [review]
Generally LGTM, Thanks!

File: src/runtime-rs/crates/shim/src/args.rs
Comment (Richardhongyu) [reply src/runtime-rs/crates/shim/src/args.rs:120]: When I first saw the code here, I thought the id 1dfc0567 has a special meaning. But it's not indeed. So I want to use a meaningful id to help the later readers.

File: src/libs/kata-types/src/config/agent.rs
Comment (Richardhongyu) [reply src/libs/kata-types/src/config/agent.rs]: Done.

**Comment by Richardhongyu** [comment]
Thanks for the reviewing and advice!

File: src/libs/kata-types/src/config/agent.rs
Comment (dcmiddle) [inline src/libs/kata-types/src/config/agent.rs:96]: nit: this field is out of order from how they are defined above. It will improve readability to list it under `enable_tracing` before `server_port`.

File: src/runtime-rs/crates/agent/src/types.rs
Comment (dcmiddle) [inline src/runtime-rs/crates/agent/src/types.rs:12]: nit: Name choice
Remove Container Request has two fields, so it would be better to specify the field in the variable name.
consider: `DEFAULT_REMOVE_CONTAINER_REQUEST_TIMEOUT`

**Review by dcmiddle — APPROVED** [review]

File: src/libs/kata-types/src/config/agent.rs
Comment (Richardhongyu) [reply src/libs/kata-types/src/config/agent.rs]: Done.

File: src/runtime-rs/crates/agent/src/types.rs
Comment (Richardhongyu) [reply src/runtime-rs/crates/agent/src/types.rs]: Done.

**Comment by Richardhongyu** [comment]
Thanks for helping reivew!

**Comment by Richardhongyu** [comment]
@liubin Hi. It seems the PR cannot run the CI test until a maintainer approves. Could you help approve? Thanks for your help in advance!

**Comment by Richardhongyu** [comment]
@liubin Hi, I fix the previous CI problem and want to run the CI test again. Thanks!

**Comment by Richardhongyu** [comment]
@studychao Hi, could you help approve the workflows? The CI test is not enabled since my last commit. Thanks for your help!

**Comment by studychao** [comment]
> @studychao Hi, could you help approve the workflows? The CI test is not enabled since my last commit. Thanks for your help!

Done！

**Comment by Richardhongyu** [comment]
> Done！

Thanks!

**Comment by Richardhongyu** [comment]
Hi, @Ji-Xinyou and @dcmiddle, I just rebase the code and pass all the `Required` CIs. Can you help review this PR again? Thanks!

There are still two CIs falling. One is due to a backport label which the recently merged PRs don't pass either. The other is because the `k8s-footloose.bats` CI always fails and is not related to this PR.

**Review by Ji-Xinyou — APPROVED** [review]
LGTM, Thanks!

**Comment by Richardhongyu** [comment]
@studychao Hi, Could you help approve the workflows? I rebased the code to solve a code conflict and it seems the CI needs to run again. BTW, if you could also help review this PR, I would be very grateful!

**Comment by studychao** [comment]
> @studychao Hi, Could you help approve the workflows? I rebased the code to solve a code conflict and it seems the CI needs to run again. BTW, if you could also help review this PR, I would be very grateful!

done.

**Comment by studychao** [comment]
@Richardhongyu commit message check failed, please take a look

**Comment by Richardhongyu** [comment]
Done. I just rebase my PR in the lastest code.

**Comment by dcmiddle** [comment]
LGTM! Hopefully that rebase makes the CI checks happy.

**Review by jodh-intel — APPROVED** [review]
Thanks @Richardhongyu.

lgtm

**Comment by jodh-intel** [comment]
@Richardhongyu - There's a tiny change the CI needs you to make. Looking at the failing CI logs:

```
15:22:12 error: unused imports: `DEFAULT_AGENT_LOG_PORT`, `DEFAULT_AGENT_VSOCK_PORT`
15:22:13   --> crates/runtimes/virt_container/src/sandbox.rs:22:15
15:22:13    |
15:22:13 22 |     default::{DEFAULT_AGENT_LOG_PORT, DEFAULT_AGENT_VSOCK_PORT},
15:22:13    |               ^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^
15:22:13    |
15:22:13    = note: `-D unused-imports` implied by `-D warnings`
15:22:13 
15:22:13 error: could not compile `virt_container` due to previous error
```

So if you can remove that line...

```diff
diff --git a/src/runtime-rs/crates/runtimes/virt_container/src/sandbox.rs b/src/runtime-rs/crates/runtimes/virt_container/src/sandbox.rs
index 25d3fcbe5..e689f05df 100644
--- a/src/runtime-rs/crates/runtimes/virt_container/src/sandbox.rs
+++ b/src/runtime-rs/crates/runtimes/virt_container/src/sandbox.rs
@@ -18,10 +18,7 @@ use common::{
 use containerd_shim_protos::events::task::TaskOOM;
 use hypervisor::{dragonball::Dragonball, Hypervisor, HYPERVISOR_DRAGONBALL};
 use kata_sys_util::hooks::HookStates;
-use kata_types::config::{
-    default::{DEFAULT_AGENT_LOG_PORT, DEFAULT_AGENT_VSOCK_PORT},
-    TomlConfig,
-};
+use kata_types::config::TomlConfig;
 use resource::{
     manager::ManagerArgs,
     network::{NetworkConfig, NetworkWithNetNsConfig},
```

... then:

- `git commit -as`
- `git rebase -i main` and squash the newly-created commit into the main one.
- `git push -f`
... we can hopefully get this landed :smile:

**Comment by jodh-intel** [comment]
Thanks @Richardhongyu !

/test

**Comment by Richardhongyu** [comment]
All the required CI items are passed.

**Review by dcmiddle — APPROVED** [review]

**Comment by dcmiddle** [comment]
thanks @jodh-intel @fidencio for getting this over the line!
thanks @Richardhongyu for sticking with it!!

**Comment by Richardhongyu** [comment]
Thank you all for your help!

**Comment by gkurz** [comment]
Let's drop the `needs-backport` label since this depends on #5788 that is too intrusive to be backported.

**Pull request merged**
